crates/transaction-pool/src/test_utils/mock.rs

```diff
@@ -660,6 +669,90 @@ impl MockTransaction {
     }
 }
 
+impl alloy_consensus::Typed2718 for MockTransaction {
```

Comment:
needed to implement `Transaction` such that `PayloadTransactionsChain` could operate on the MockTransaction type.

Need to update some of these stubbed out methods